src/Cake.Core/Polyfill/EnvironmentHelper.cs

```diff
@@ -40,8 +40,9 @@ public static PlatformFamily GetPlatformFamily()
                 return PlatformFamily.Windows;
             }
 #else
+            var isRunningOnMac = Native.MacOSX.IsRunningOnMac();
             var platform = (int)Environment.OSVersion.Platform;
-            if (platform == (int)PlatformID.MacOSX)
+            if (isRunningOnMac || platform == (int)PlatformID.MacOSX)
```

Comment:
Would it be possible to add this line last in the checks to avoid the call on Windows and Linux? Otherwise, would it be possible to cache the result.

Also, are we sure this won't throw on Windows and Linux?